README.md

```diff
@@ -1,229 +1,894 @@
+call stack in this example, so just ignore it in this case. Stack trace is
+always emitted by `retsnoop`, there is no way to suppress it. This might be
+possible in the future.
+
+Back to function call trace table. Let's shorten it some more to explain main
+elements:
 
-```shell
-$ sudo retsnoop -e '*_sys_write' -a ':fs/btrfs/*.c' -v
-[...]
+```
+FUNCTION CALLS TRACE                              RESULT                 DURATION
+-----------------------------------------------   --------------------  ---------
+→ bpf_prog_load
+    → bpf_prog_alloc
+        ↔ bpf_prog_alloc_no_stats                 [0xffffc9000031e000]    5.539us
+    ← bpf_prog_alloc                              [0xffffc9000031e000]   10.265us
+    [...]
+    → bpf_prog_kallsyms_add
+        ↔ bpf_ksym_add                            [void]                  2.046us
+    ← bpf_prog_kallsyms_add                       [void]                  6.104us
+← bpf_prog_load                                   [5]                   374.697us
 ```
 
-## Use cases
+In the left column we see function names with one of `→`, `←`, or `↔` markers
+on the left. `→` marks entry into traced function, `←` marks and exit from a
+function. If a function didn't have any further calls to any other traced
+function and thus is a leaf function (as far as a specified set of functions of
+interest goes; note, that in reality there could have been other functions
+called further, but they were not traced and thus captured by `retsnoop` due to
+them being omitted from `-e`/`-d` specifications). Such leaf functions, instead
+of taking two rows with separate `→` (entry) and `←` (exit) records, are
+collapsed into a single entry-exit row marked with `↔`. As calls become more
+nested, `retsnoop` indents each level of nestedness with four spaces.
+
+For each row corresponding to function exit event (so, `→`- and `↔`-marked
+rows), retsnoop also reports return result, similarly to default stack trace
+view. And also similarly to stack trace view, we get a duration of a complete
+function call (inclusive of all the child functions' execution time as well).
+
+So from short excerpt above, we can see that `bpf_prog_load` was called, it
+then called `bpf_prog_alloc` which further called `bpf_prog_alloc_no_stats`.
+`bpf_prog_alloc_no_stats` returned some valid kernel address
+(`0xffffc9000031e000`), which was subsequently returned from `bpf_prog_alloc`
+back to `bpf_prog_load`. We then skip a lot of function invocations for
+brevity, but eventually we get to `bpf_prog_kallsyms_add`, which calls
+`bpf_ksym_add` once. Then `bpf_ksym_add` and, subsequently,
+`bpf_prog_kallsyms_add` return with no results (they are `void` functions,
+according to BTF type information). After `bpf_prog_kallsyms_add` returned,
+`bpf_prog_load` completes and returns a successful result (FD 5 in this case),
+entire invocation of `bpf_prog_load` takes almost 375 microseconds.
```

Comment:
suggestion: I'd make this last line a separate sentence, starting with "The entire invocation..."